elements/lisk-transactions/src/utils/verify.ts

```diff
@@ -54,72 +53,89 @@ export const verifyMinRemainingBalance = (
 	return undefined;
 };
 
-export interface VerifyMultiSignatureResult {
-	readonly status: MultisignatureStatus;
-	readonly errors: ReadonlyArray<TransactionError>;
-}
-
-const isMultisignatureAccount = (account: Account): boolean =>
+export const isMultisignatureAccount = (account: Account): boolean =>
 	!!(
 		(account.keys.mandatoryKeys.length > 0 ||
 			account.keys.optionalKeys.length > 0) &&
 		account.keys.numberOfSignatures
 	);
 
-export const verifyMultiSignatures = (
+export const validateKeysSignatures = (
+	keys: readonly string[],
+	signatures: readonly string[],
+	transactionBytes: Buffer,
+) => {
+	const errors = [];
+	// tslint:disable-next-line: prefer-for-of no-let
+	for (let i = 0; i < keys.length; i += 1) {
+		const { error } = validateSignature(
+			keys[i],
+			signatures[i],
+			transactionBytes,
+		);
+
+		if (error) {
+			errors.push(error);
+		}
+	}
+
+	return errors;
+};
+
+export const verifyMultiSignatureTransaction = (
 	id: string,
 	sender: Account,
 	signatures: ReadonlyArray<string>,
 	transactionBytes: Buffer,
-): VerifyMultiSignatureResult => {
-	if (!isMultisignatureAccount(sender) && signatures.length > 0) {
-		return {
-			status: MultisignatureStatus.FAIL,
-			errors: [
-				new TransactionError(
-					'Sender is not a multisignature account',
-					id,
-					'.signatures',
-				),
-			],
-		};
+) => {
+	const errors = [];
+	if (signatures.length === 0) {
+		const error = new TransactionError(
+			'Transaction has empty signatures',
+			id,
+			'.signatures',
+		);
+
+		return [error];
 	}
 
-	if (!isMultisignatureAccount(sender)) {
-		return {
-			status: MultisignatureStatus.NONMULTISIGNATURE,
-			errors: [],
-		};
+	const { mandatoryKeys, optionalKeys, numberOfSignatures } = sender.keys;
+	const numMandatoryKeys = mandatoryKeys.length;
+	const nonEmptySignaturesCount = signatures.filter(k => k.length !== 0).length;
+
+	if (nonEmptySignaturesCount !== numberOfSignatures) {
+		const error = new TransactionError(
+			`Transaction signatures does not match required number of transactions: ${numberOfSignatures}`,
+			id,
+			'.signatures',
+			signatures.join(','),
+		);
+
+		return [error];
 	}
 
-	const { valid, errors } = validateMultisignatures(
-		[], // UPDATE TO LIP0017 sender.membersPublicKeys as ReadonlyArray<string>,
+	const numOptionalKeys = optionalKeys.length;
+
+	const mandatoryKeysError = validateKeysSignatures(
+		mandatoryKeys,
 		signatures,
-		sender.keys.numberOfSignatures,
 		transactionBytes,
-		id,
 	);
 
-	if (valid) {
-		return {
-			status: MultisignatureStatus.READY,
-			errors: [],
-		};
-	}
+	errors.push(...mandatoryKeysError);
 
-	if (
-		errors &&
-		errors.length === 1 &&
-		errors[0] instanceof TransactionPendingError
-	) {
-		return {
-			status: MultisignatureStatus.PENDING,
-			errors,
-		};
+	// tslint:disable-next-line: prefer-for-of no-let
+	for (let k = 0; k < numOptionalKeys - 1; k += 1) {
```

Comment:
Can you add comments about why are you modifying the limit (`< numOptionalKeys - 1`) and what's the objective of the whole loop?